tests/src/case0054.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2009-2020, Christian Ferrari <tiian@users.sourceforge.net>
+ * All rights reserved.
+ *
+ * This file is part of LIXA.
+ *
+ * LIXA is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation.
+ *
+ * LIXA is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LIXA.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <config.h>
+
+
+
+#ifdef HAVE_STDIO_H
+# include <stdio.h>
+#endif
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif
+#ifdef HAVE_ASSERT_H
+# include <assert.h>
+#endif
+
+
+
+#include <tx.h>
+#include <lixa_tx.h>
+#include <liblixamonkey.h>
+
+
+
+/* this is a special case test: it is a basic test for statically registered
+   resource managers with a code-specified profile supplied as an argument */
+
+
+
+int main(int argc, char *argv[])
+{
+    char *pgm = argv[0];
+    char *profile = NULL;
+    int rc;
+    TXINFO info;
+
+    if (argc > 0) {
+        profile = argv[1];
+    }
+    
+    printf("%s| starting...\n", pgm);
+    printf("%s| lixa_tx_set_profile(%s): %d\n", pgm, profile ? profile : "NULL", rc = lixa_tx_set_profile(profile));
```

Comment:
To stay aligned with the TX Demarcation Specification, a "tx_set_profile" must be put in "tx.h" as for the other functions. "lixa_tx_*" must be considered an internal iterface. Look for example at "tx_set_transaction_control" in tx.h and tx.c files.